src/util/urlHelper.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2017-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { matchPath } from 'react-router-dom';
+import Url from 'url-parse';
+import { match as RouterMatchType } from 'react-router';
+import { isValidLocale } from '../i18n';
+import {
+  PLAIN_ARTICLE_PAGE_PATH,
+  RESOURCE_PAGE_PATH,
+  TOPIC_PATH,
+  PLAIN_ARTICLE_IFRAME_PATH,
+  RESOURCE_ARTICLE_IFRAME_PATH,
+} from '../constants';
+
+function matchUrl<Params>(
+  pathname: string,
+  lang: boolean = false,
+): RouterMatchType<Params> | null {
+  const possiblePaths = [
+    lang ? `/:lang${RESOURCE_PAGE_PATH}` : RESOURCE_PAGE_PATH,
+    lang ? `/:lang${TOPIC_PATH}` : TOPIC_PATH,
+    lang ? `/:lang${PLAIN_ARTICLE_PAGE_PATH}` : PLAIN_ARTICLE_PAGE_PATH,
+    lang ? `/:lang${PLAIN_ARTICLE_IFRAME_PATH}` : PLAIN_ARTICLE_IFRAME_PATH,
+    lang
+      ? `/:lang${RESOURCE_ARTICLE_IFRAME_PATH}`
+      : RESOURCE_ARTICLE_IFRAME_PATH,
+  ];
+
+  for (let i = 0; i < possiblePaths.length; i++) {
+    const attempt = possiblePaths[i] ?? '';
+    const m = matchPath<Params>(pathname, attempt);
+    if (m) {
+      return m;
+    }
+  }
```

Comment:
tips for å gjøre denne kulere uten å kalle `matchPath` flere ganger enn nødvendig tas imot med takk :smile: 
I scala hadde jeg brukt `collectFirst`, men klarte ikke finne noe tilsvarende i js/ts(?)

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find